src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/UnionAllExpression.cs

```diff
@@ -0,0 +1,69 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Expressions
+{
+    /// <summary>
+    /// Represents a union all operator.
+    /// </summary>
+    public class UnionAllExpression : Expression, IExpressionsContainer
```

Comment:
UnionAll still sounds very specific for an expression, in most other cases here it would be `UnionExpression` with property `Type.All`. Or is this just a variant of `MultiaryExpression` where the MultiaryOperation is `Union`?